packages/compiler-sfc/src/compileScript.ts

```diff
@@ -1065,11 +1066,32 @@ export function compileScript(
           if (isFunctionType(child)) {
             this.skip()
           }
+          if (child.type === 'ExpressionStatement') {
```

Comment:
The logic is correct but the implementation is a bit messy - I refactored it in be6a0bff

In the future, please [tick the "allow edits from maintainer" when submitting PRs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) so that we can refactor in the PR itself instead of creating separate commits.

Reply:
thank you! ok i see.